client/me/purchases/list/index.jsx

```diff
@@ -55,24 +55,32 @@ const PurchasesList = React.createClass( {
 		);
 	},
 
-	render() {
-		let content;
+	isDataLoading() {
+		if ( this.props.purchases.isFetching && ! this.props.purchases.hasLoadedFromServer ) {
+			return true;
+		}
 
-		if ( ! this.props.purchases.isFetching && ! this.props.purchases.hasLoadedFromServer ) {
-			return null;
+		if ( ! this.props.sites.fetched ) {
```

Comment:
What about checking against `sites.initialized`? It looks like it works the same way as `sites.fetched`, but also works for data cached in local storage.